### PR TITLE
dts: binding: refine nxp rdc property

### PR DESCRIPTION
**Pull request by JiafeiPan**
Define rdc property in a yaml file and include it in the peripheral's dts binding.

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

File: dts/bindings/misc/nxp,rdc-policy.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  rdc:
+    type: int
+    description: |
+      Set the RDC permission for specified peripheral: Resource Domain Controller (RDC)
+      supports flexible configuration on IP’s access permission, each individual IP can,
+      for example, be configured as A-core only or M-core only. This property uses some
+      macros defined in imx_rdc.h to configure current peripheral's access permission.
+      For example to allow both A53 and M7 Core to have read/write permission to current
+      peripheral:
+      rdc = <(RDC_DOMAIN_PERM(A53_DOMAIN_ID, RDC_DOMAIN_PERM_RW)|\
```
Comment (decsny) [inline dts/bindings/misc/nxp,rdc-policy.yaml:14]: what I would like to see is all the options of the macros listed in this description

**Review by decsny — CHANGES_REQUESTED** [review]

File: dts/bindings/misc/nxp,rdc-policy.yaml (hunk 1 above)
Comment (manuargue) [reply dts/bindings/misc/nxp,rdc-policy.yaml:14]: I don't think that would be necessary, as the attributes can be often SoC specific (e.g. not all devices define the same domains and partitions). The current change is somewhat similar to what is already done for other types of attributes like https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/base/zephyr%2Cmemory-common.yaml#L22-L27

File: dts/bindings/misc/nxp,rdc-policy.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+properties:
+  rdc:
+    type: int
+    description: |
+      Set the RDC permission for specified peripheral: Resource Domain Controller (RDC)
+      supports flexible configuration on IP’s access permission, each individual IP can,
+      for example, be configured as A-core only or M-core only. This property uses some
+      macros defined in imx_rdc.h to configure current peripheral's access permission.
```
Comment (manuargue) [inline dts/bindings/misc/nxp,rdc-policy.yaml:11]:
```suggestion
      macros defined in 'include/zephyr/dt-bindings/rdc/imx_rdc.h' to configure current peripheral's access permission.
```

File: dts/bindings/misc/nxp,rdc-policy.yaml (hunk 1 above)
Comment (JiafeiPan) [reply dts/bindings/misc/nxp,rdc-policy.yaml:15]: Updated, thanks.

File: dts/bindings/misc/nxp,rdc-policy.yaml (hunk 2 above)
Comment (JiafeiPan) [reply dts/bindings/misc/nxp,rdc-policy.yaml]: updated, thanks.

**Comment by JiafeiPan** [comment]
Rebased and updated dts binding description.

**Review by manuargue — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

File: dts/bindings/misc/nxp,rdc-policy.yaml (hunk 1 above)
Comment (decsny) [reply dts/bindings/misc/nxp,rdc-policy.yaml:15]: well, the reference to the file with the macros is something, I guess...

**Review by decsny — APPROVED** [review]

File: dts/bindings/misc/nxp,rdc-policy.yaml (hunk 1 above)
Comment (decsny) [reply dts/bindings/misc/nxp,rdc-policy.yaml:15]: > I don't think that would be necessary, as the attributes can be often SoC specific (e.g. not all devices define the same domains and partitions). The proposed change on this pr is somewhat similar to what is already done for other types of attributes like https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/base/zephyr%2Cmemory-common.yaml#L22-L27

something is necessary, user when looking at binding documentation should not have to guess

**Review by dleach02 — APPROVED** [review]

**Pull request merged**